src/coreclr/pal/src/thread/context.cpp

```diff
@@ -2181,6 +2184,8 @@ DBG_FlushInstructionCache(
 #endif
 
     syscall(__NR_riscv_flush_icache, (char *)lpBaseAddress, (char *)((INT_PTR)lpBaseAddress + dwSize), 0 /* all harts */);
+#elif defined(HOST_APPLE) && !defined(HOST_OSX)
+    sys_icache_invalidate((void *)lpBaseAddress, dwSize);
```

Comment:
Can we use this on all Apple platforms?

Reply:
Probably yes. I'll check.

Reply:
It seems that Clang optimizes/inlines `sys_icache_invalidate` on macOS so there's no additional overhead to using it. The same pattern is already present in NativeAOT PAL. I can submit a separate small PR to fix/unify both of these places.

Reply:
I have just tried to compile a simple testing C app with `sys_icache_invalidate` and `__builtin___clear_cache` on my M1 mac. It actually didn't inline anything, and the `__builtin___clear_cache` code just tail called `sys_icache_invalidate`. 
However, on mac x64, the `__builtin___clear_cache` compiles to nothing while the `sys_icache_invalidate` is still a call.

Reply:
> However, on mac x64, the `__builtin___clear_cache` compiles to nothing while the `sys_icache_invalidate` is still a call.

That's what I originally expected and why I left the `__builtin___clear_cache` in place.

I misinterpreted the output of my test case because `objdump -S` didn't print the correct method names in the object file.

Reply:
Let's leave this change as is then.